src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java

```diff
@@ -65,8 +67,9 @@ private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
 
-    public boolean hasSolicitorRole(List<String> roles) {
-        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches());
+    public boolean hasSolicitorRoleForJurisdiction(List<String> roles, String jurisdiction) {
+        String jurisdictionRole = String.format(CASEWORKER_JURISDICTION, jurisdiction).toLowerCase(Locale.getDefault());
+        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches()
+            && role.toLowerCase(Locale.getDefault()).startsWith(jurisdictionRole));
```

Comment:
The logic is slightly different to data-store-api but works fine as long as below multi jurisdiction scenario never exist. Could you check with Richard once please. 
role : `caseworker-divorce-financialremedy-solicitor` but jurisdiction : financialremedy ??

Reply:
Will check with Richard. I had thought that the jurisdiction in the above example was divorce?